### ISSUE TITLE
Enhancement to AWS::EC2::SecurityGroup Linting - CidrIp or SourceSecurityGroupId

### ISSUE THREAD

**Issue by MattHodge** [awslabs/cfn-python-lint#251]
Hi there 👋 

*cfn-lint version: 0.4.2*

*Description of issue.*

Given the following resource:

```yaml
  SecurityGroup:
    Type: AWS::EC2::SecurityGroup
    Properties:
      VpcId: !Ref VPCID
      SecurityGroupIngress:
        - CidrIp: 1.1.1.1/24
          FromPort: 123
          IpProtocol: tcp
          SourceSecurityGroupId: Something
          ToPort: 3389
```

Stack deploy will fail with `Both CidrIp and SourceSecurityGroup cannot be specified`

**Comment by MattHodge** [comment]
Wait it does actually warn about this!
